src/index.ts

```diff
@@ -0,0 +1,69 @@
+import * as chalk from "chalk";
+import * as tty from "tty";
+
+
+const emptyDisposable = {
+    dispose: () => { return; }
+};
+
+/**
+ * Object that releases resources when "dispose" is invoked.
+ */
+export interface Disposable {
+    /**
+     * Releases any resources held by the implementing object.
+     */
+    dispose(): void;
+}
+
+/**
+ * Prints a message indicating Ctrl+C was pressed then kills the process with SIGINT status.
+ */
+export function defaultCtrlCHandler(): void {
+    console.log(`'${chalk.cyan("^C")}', exiting`);
+    process.kill(process.pid, "SIGINT");
+}
+
+/**
+ * Monitors Ctrl+C and executes a callback instead of SIGINT.
+ *
+ * When invoked, this listens to keypresses on STDIN,
+ * which prevents normal process termination. To ensure
+ * your program terminates properly if Ctrl+C is NOT
+ * pressed, call the "dispose" method on the object
+ * returned by this function when your process is
+ * completed.
+ *
+ * NOTE: This should only be used by programs that do
+ * not normally read from STDIN.
+ *
+ * @param {Function} onCtrlC
+ *     Callback function to execute on Ctrl+C.
+ *     @default "defaultCtrlCHandler"
+ * @returns
+ *     Disposable object that restores STDIN to its previous
+ *     state, allowing for proper process termination.
+ */
+export function monitorCtrlC(onCtrlC: Function = defaultCtrlCHandler): Disposable {
```

Comment:
I'd rather this type be more specific: `(onCtrlC: () => any = defaultCtrlCHandler)